### DISCUSSION TITLE
[4.0] Equalize impressions between banner clients

### DISCUSSION THREAD

**Discussion by wesleyferreiraadv** [joomla/joomla-cms#39243] [2. Cold Features]
### Steps to reproduce the issue
It inserted 3 Banners clients (Client A, B and C). Assign 3 banner to client A, 1 to client B and 1 to Client C). Create a mod_banner that retrieves 1 banner at a time, randomly. Generate some impressions on the frontend. Upon returning to the back end, Client A (with more banners) will have 3 times more impressions than the others.

### Expected result
When performing the random operation of selecting the banner to be displayed, consider for the lottery the customers and not the registered banners, in general.

### Actual result
Currently, a customer who has more banner than others receives more impressions, even though they have contracted the same ad package.

### System information (as much as possible)
In the scenario named, the joomla native com_banner and mod_banner were used, with no changes or third-party components.

### Additional comments
I have a news portal. Seeing ads for merchants. The amounts paid by them are identical. Each advertiser can register varied banners (selling clothes, shoes, toys) and the amount of banners between clients is different. Advertisers who send more banners are receiving proportionately more exposure than those who send fewer banners. This has brought problems. The ideal scenario is that if there are 3 customers, the random draw first select a customer and then a customer banner selected so that the impressions are fairly distributed.

**Comment by ghost** [comment]
@Quy can you please have a Look?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24175">issues.joomla.org/tracker/joomla-cms/24175</a>.</sub>

**Comment by HLeithner** [comment]
It's not easy to achieve this with our current banner selection code, you can request this as feature for 4.0 and find someone who wants to write a PR.

edit: thx brian

**Comment by ghost** [comment]
@drwesleyferreira please add "[4.0] " at Start of the Title.

**Comment by alikon** [comment]
let's move it to discussion

**Reply by crystalenka** [reply]
Converted back to issue.
